### PR TITLE
Login

### PR DESCRIPTION
**Pull request by zuzanna81**
### Description
- Added feature and scenarios for login feature
- Added step definitions for login feature

### CODE REVIEW

File: features/step_definitions/login_steps.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  click_link_or_button element
+end
+
+Then("I should be redirected to the {string} page") do |page_name|
+  expect(page.current_path).to eq page_path_from(page_name)
+end
+
+Then("I should be logged in as {string}") do |email|
+  user = User.find_by(email: email)
+  login_as(user, scope: :user)
```
Comment (tochman) [inline features/step_definitions/login_steps.rb]: This is not an assertion. You are actually logging in that user with the `login_as(user, scope: :user)` so there's no way in knowing if your scenario has passed

**Review by tochman — DISMISSED** [review]
Looks good apart from one thing.... See comment.

File: features/step_definitions/login_steps.rb (hunk 1 above)
Comment (ScottyyyB) [reply features/step_definitions/login_steps.rb]: Okay that was my bad, will fix it asap :)

**Review by NegativeKarma — DISMISSED** [review]
LGTM

File: features/step_definitions/login_steps.rb (hunk 1 above)
Comment (ScottyyyB) [reply features/step_definitions/login_steps.rb]: Modified my step definition, not sure if it's exactly what you wanted, was the closest I could find to an assertion to check that the specific user has signed in.

File: features/step_definitions/login_steps.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  click_link_or_button element
+end
+
+Then("I should be redirected to the {string} page") do |page_name|
+  expect(page.current_path).to eq page_path_from(page_name)
+end
+
+Then("I should be logged in as {string}") do |email|
+  user = User.find_by(email: email)
+  expect(user.sign_in_count).to eq 1
```
Comment (diraulo) [inline features/step_definitions/login_steps.rb]: Just wondering if this is enough to ensure that someone has successfully logged in 🤔 
Don't think you have access to `current_user` in cucumber steps, but one way I'd ensure that someone correctly logged in would be to information unique to them displayed on the page after login (i.e `Welcome user@email.tld`)

Comment (ScottyyyB) [reply features/step_definitions/login_steps.rb]: Yeah true, I have that added in my dynamic navbar feature, where whenever a user is signed it displays "Logged in as, their_email@gmail.com", so could test with that instead as you said when the other PR gets merged.

Comment (ScottyyyB) [reply features/step_definitions/login_steps.rb]: So just having a simple step like so should be fine then right: Then I should see "Logged in as, their_email@gmail.com"

Comment (NegativeKarma) [reply features/step_definitions/login_steps.rb]: ye, you basically already add that here https://github.com/CraftAcademy/royale-with-cheese/pull/6/files#diff-86d21891955ee1c61f8dfa0892a37bb0  @diraulo

**Review by diraulo — APPROVED** [review]
This looks good to me...

**Pull request merged**